Service/DataService.php

```diff
@@ -2010,4 +2023,101 @@ public function createAndMapIndex(Environment $environment): void
             'name' => $environment->getAlias()
         ]);
     }
+
+    public function getDataLinks(string $contentTypesCommaList, array $businessIds): array
+    {
+        $items = [];
+        $ouuids = [];
+        foreach ($businessIds as $businessId) {
+            if (isset($this->cacheOuuids[$contentTypesCommaList][$businessId])) {
+                $ouuids[$businessId] = $this->cacheOuuids[$contentTypesCommaList][$businessId];
+            } else {
+                $items[] = $businessId;
+                $ouuids[$businessId] = $businessId;
+            }
+        }
+
+        foreach ($this->contentTypeService->getByNames(explode(',', $contentTypesCommaList)) as $contentType) {
+            if ($contentType->getBusinessIdField() && count($ouuids) > 0) {
+                $result = $this->client->search([
+                    'index' => $contentType->getEnvironment()->getAlias(),
+                    'body' => [
+                        'size' => sizeof($ouuids),
+                        '_source' => $contentType->getBusinessIdField(),
+                        'query' => [
+                            'bool' => [
+                                'must' => [
+                                    [
+                                        'term' => [
+                                            '_contenttype' => $contentType->getName()
+                                        ]
+                                    ],
+                                    [
+                                        'terms' => [
+                                            $contentType->getBusinessIdField() => $items
+                                        ]
+                                    ],
+                                ]
+                            ]
+                        ]
+
+                    ],
+                    'size' => 100,
+                    "scroll" => "1m"
+                ]);
+
+                while (count($result['hits']['hits'] ?? []) > 0) {
+                    foreach ($result['hits']['hits'] as $hits) {
+                        $key = sprintf('%s:%s', $contentType->getName(), $hits['_id']);
+                        $ouuids[$hits['_source'][$contentType->getBusinessIdField()]] = $key;
+                        $this->cacheOuuids[$contentTypesCommaList][$contentType->getBusinessIdField()] = $key;
+                    }
+                    $result = $this->client->scroll([
+                        'scroll_id' => $result['_scroll_id'],
+                        'scroll' =>  "1m",
+                    ]);
+                }
+            }
+        }
+        return array_values($ouuids);
+    }
+
+    public function getDataLink(string $contentTypesCommaList, string $businessId): ?string
+    {
+        return $this->getDataLinks($contentTypesCommaList, [$businessId])[0] ?? $businessId;
```

Comment:
Not a fan of this kind of constructions. get DataLinks should return an object, not an array, that way you can ask the first result. getDataLink might return 10 results, forcing the first one without an explicit "->first()" call on an object is really hard to understand for other developers.